lib/android/app/src/main/java/com/reactnativenavigation/presentation/FabPresenter.kt

```diff
@@ -0,0 +1,194 @@
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import com.google.android.material.floatingactionbutton.FloatingActionButton.SIZE_MINI
+import com.google.android.material.floatingactionbutton.FloatingActionButton.SIZE_NORMAL
+import com.reactnativenavigation.R
+import com.reactnativenavigation.interfaces.FloatingButton
+import com.reactnativenavigation.parse.FabOptions
+import com.reactnativenavigation.parse.Options
+import com.reactnativenavigation.utils.ObjectUtils
+import com.reactnativenavigation.viewcontrollers.ViewController
+import com.reactnativenavigation.views.ExtendedFab
+import com.reactnativenavigation.views.Fab
+
+
+class FabPresenter {
+    private var viewGroup: ViewGroup? = null
+    private var component: ViewController<*>? = null
+    private var fab:FloatingButton? = null
+
+    private fun createFab(options: Options) {
+        if (options.extendedFabOptions.hasValue()) {
+            fab = ExtendedFab(viewGroup!!.context, options.fabOptionsType.id.get())
+        } else {
+            fab = Fab(viewGroup!!.context, options.fabOptionsType.id.get())
+        }
+        with(options.fabOptionsType) {
+            setParams(fab!! as View, this)
+            applyFabOptions(fab!!, this)
+            fab!!.setOnClickListener { _: View? -> component!!.sendOnNavigationButtonPressed(this.id.get()) }
+        }
+
+        viewGroup!!.addView(fab as View)
+    }
+
+    fun applyOptions(options: Options, component: ViewController<*>, viewGroup: ViewGroup) {
+        this.viewGroup = viewGroup
+        this.component = component
+        with(options.fabOptionsType) {
+            if (this.id.hasValue()) {
+                if (fab != null && fab!!.fabId == this.id.get()) {
+                    fab!!.bringToFront()
+                    applyFabOptions(fab!!, this)
+                    setParams(fab!! as View, this)
+                    fab!!.setOnClickListener { _: View? -> component.sendOnNavigationButtonPressed(this.id.get()) }
+                } else {
+                    createFab(options)
+                }
+            } else {
+                removeFab()
+            }
+        }
+    }
+
+    fun mergeOptions(options: Options, component: ViewController<*>, viewGroup: ViewGroup) {
+        this.viewGroup = viewGroup
+        this.component = component
+        with(options.fabOptionsType) {
+            if (this.id.hasValue()) {
+                if (fab != null && fab!!.fabId == this.id.get()) {
+                    mergeParams(fab!! as View, this)
+                    fab!!.bringToFront()
+                    mergeFabOptions(fab!!, this)
+                    fab!!.setOnClickListener { _: View? -> component.sendOnNavigationButtonPressed(this.id.get()) }
+                } else {
+                    createFab(options)
+                }
+            }
+        }
+    }
+
+    private fun removeFab() {
+        if (fab != null) {
```

Comment:
Kotlin simplifies working with nulls with the `?.let` syntax. So instead of checking for nullability, a more readable syntax would be:

```kotlin
fab?.let {
  it.hide()
  ...
}
```